src/azure-cli/azure/cli/command_modules/storage/operations/fs_file.py

```diff
@@ -0,0 +1,139 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+"""Custom operations for storage file datalake"""
+
+import os
+
+from azure.core.exceptions import HttpResponseError
+from azure.cli.core.profiles import ResourceType
+from azure.cli.command_modules.storage.util import mkdir_p
+from knack.log import get_logger
+from knack.util import CLIError
+
+logger = get_logger(__name__)
+
+
+def append_file(client, content, timeout=None):
+    file = client.get_file_properties(timeout=timeout)
+    # START append data
+    client.append_data(data=content, offset=file.size, length=len(content), timeout=timeout)
+    # END append data
+    return client.flush_data(offset=file.size + len(content), timeout=timeout)
+
+
+def download_file(client, destination_path=None, overwrite=True, timeout=None):
+    destination_folder = os.path.dirname(destination_path) if destination_path else ""
+    if destination_folder and not os.path.exists(destination_folder):
+        mkdir_p(destination_folder)
+
+    if not destination_folder or os.path.isdir(destination_path):
+        file = client.get_file_properties(timeout=timeout)
+        file_name = file.name.split("/")[-1]
+        destination_path = os.path.join(destination_path, file_name) \
+            if destination_path else file_name
+
+    if not overwrite and os.path.exists(destination_path):
+        raise CLIError('The specified path already exists. Please change to a valid path. ')
+
+    with open(destination_path, 'wb') as stream:
+        download = client.download_file(timeout=timeout)
+        download_content = download.readall()
+        stream.write(download_content)
+
+
+def exists(cmd, client, timeout=None):
+    try:
+        client.get_file_properties(timeout=timeout)
+        return True
+    except HttpResponseError as ex:
+        from azure.cli.command_modules.storage.track2_util import _dont_fail_on_exist
+        StorageErrorCode = cmd.get_models("_shared.models#StorageErrorCode",
+                                          resource_type=ResourceType.DATA_STORAGE_FILEDATALAKE)
+        _dont_fail_on_exist(ex, StorageErrorCode.blob_not_found)
+        return False
```

Comment:
same as above

Reply:
to catch the release, I will merge it first and we can consider the exception behavior for exist() later.